torch_xla/csrc/ops/dequant_tensor.cpp

```diff
@@ -36,13 +36,15 @@ torch::lazy::NodePtr DequantizeTensor::Clone(
 XlaOpVector DequantizeTensor::Lower(LoweringContext* loctx) const {
   xla::XlaOp input = loctx->GetOutputOp(operand(0));
   xla::Shape input_shape = ShapeHelper::ShapeOfXlaOp(input);
+  xla::Shape output_shape = xla::ShapeUtil::MakeShape(xla::PrimitiveType::F32,
```

Comment:
Should we assert that the input_shape uses f32 storage type? Is this guaranteed?

Reply:
Edit: I see that dequantize decomposes to f32 in PT. Probably OK for now. Not sure if there's a way to future proof this. I.e. can we query the storage type from elsewhere? The PT op maybe?

Reply:
Unfortunately the PT op like the followings:
```
x1 = torch.ops.quantized_decomposed.quantize_per_tensor(
    x, 0.4, 2, -128, 127, torch.int8)
x2 = torch.ops.quantized_decomposed.dequantize_per_tensor(
    x1, 0.4, 2, -128, 127, torch.int8)
```

does not convey the expressed type (`f32` in this case) as its attributes. For quantize PT op, it is determined via the input argument `x` and for de-quantize the expressed type can be determined via the input argument of the operation generating `x1`.  The fact that quantize decomposes to f32 in PT readily provides us output type information for dequantize.